x-pack/plugins/enterprise_search/public/applications/app_search/components/document_creation/creation_mode_components/paste_json_text.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { useValues, useActions } from 'kea';
+
+import { i18n } from '@kbn/i18n';
+import {
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiTextArea,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+
+import { AppLogic } from '../../../app_logic';
+import { ConfiguredLimits } from '../../../types';
+
+import { MODAL_CANCEL_BUTTON, MODAL_CONTINUE_BUTTON } from '../constants';
+import { DocumentCreationLogic } from '../';
+
+import './paste_json_text.scss';
+
+export const PasteJsonText: React.FC = () => (
+  <>
+    <ModalHeader />
+    <ModalBody />
+    <ModalFooter />
+  </>
+);
+
+export const ModalHeader: React.FC = () => {
+  return (
+    <EuiModalHeader>
+      <EuiModalHeaderTitle>
+        <h2>
+          {i18n.translate('xpack.enterpriseSearch.appSearch.documentCreation.pasteJsonText.title', {
+            defaultMessage: 'Create documents',
+          })}
+        </h2>
+      </EuiModalHeaderTitle>
+    </EuiModalHeader>
+  );
+};
+
+export const ModalBody: React.FC = () => {
+  const { configuredLimits } = useValues(AppLogic);
+  const {
+    engine: { maxDocumentByteSize },
+  } = configuredLimits as ConfiguredLimits;
```

Comment:
It looks like you cast this here to avoid handling the `undefined` case for `AppLogic.configuredLimits`. Is this OK? If the network is slow or something, is actually possible for this data to not exist when this code executes? IIRC correctly, the Enterprise Search plugin will not load unless the initial fetch for this data is not made, so it should be safe. Just checking.

Reply:
Yeah, it's not realistically possible for it to not exist due to network slowness. We grab configured limits from the initial `config_data` XHR which checks and sets things like Enterprise Search's URL, read only mode, etc. - if that failed the entire app gets kicked to the "Cannot connect" error screen. It might theoretically be possible for an App Search instance to return malformed server data though...

On second look I'm not a huge fan of this either, what do you think about this as an alternative?

```js
  const { configuredLimits } = useValues(AppLogic);
  const maxDocumentByteSize = configuredLimits?.engine?.maxDocumentByteSize;
```

Reply:
dda2ff9 - hopefully that looks good to you - if not I can revert

Reply:
Yeah that looks good.